tensorflow_datasets/core/download/kaggle.py

```diff
@@ -129,56 +112,27 @@ def __init__(self, competition_name):
     self._kaggle_type = _get_kaggle_type(self._competition_name)
 
   @utils.memoized_property
-  def competition_files(self):
-    """List of competition files."""
-    command = [
-        "kaggle",
-        self._kaggle_type.download_cmd,
-        "files",
-        "-v",
-        self._competition_name,
-    ]
-    output = _run_kaggle_command(command, self._competition_name)
-    return sorted([
-        line.split(",")[0] for line in output.split("\n")[1:] if line
-    ])
+  def competition_url(self):
+    """Returns 'kaggle.com' urls."""
+    return KaggleFile(self._competition_name).to_url()
 
-  @utils.memoized_property
-  def competition_urls(self):
-    """Returns 'kaggle://' urls."""
-    return [
-        KaggleFile(self._competition_name, fname).to_url()
-        for fname in self.competition_files  # pylint: disable=not-an-iterable
-    ]
-
-  def download_file(self, fname, output_dir):
+  def download_competition(self, competition, output_dir):
     """Downloads competition file to output_dir."""
-    if fname not in self.competition_files:  # pylint: disable=unsupported-membership-test
-      raise ValueError("%s is not one of the competition's "
-                       "files: %s" % (fname, self.competition_files))
-    command = [
-        "kaggle",
-        self._kaggle_type.download_cmd,
-        "download",
-        "--file",
-        fname,
-        "--path",
-        output_dir,
-        self._kaggle_type.dl_flag,
-        self._competition_name
-    ]
-    if self._kaggle_type.extra_flag:
-      command.append(self._kaggle_type.extra_flag)
+    command = ["kaggle",
+               self._kaggle_type.download_cmd,
+               "download",
+               self._kaggle_type.dl_flag,
+               competition,
+               "-p",
+               output_dir]
     _run_kaggle_command(command, self._competition_name)
-    # kaggle silently compresses some files to '.zip` files.
-    # TODO(tfds): use --unzip once supported by kaggle
-    # (https://github.com/Kaggle/kaggle-api/issues/9)
-    fpath = os.path.join(output_dir, fname + ".zip")
-    if zipfile.is_zipfile(fpath):
-      e = extractor.get_extractor()
-      with e.tqdm():
-        e.extract(fpath, resource.ExtractMethod.ZIP, output_dir).get()
-    return os.path.join(output_dir, fname)
+    downloads = tf.io.gfile.listdir(output_dir)
+    if len(downloads) != 1:
+      zipper = zipfile.ZipFile(os.path.join(output_dir, 'zipped.zip'))
+      for download in downloads:
+        zipper.write(os.path.join(output_dir, download))
+      zipper.close()
```

Comment:
Contextmanager should be highly preferred to manual `.close()` call.

Reply:
Fixed